src/pytest/__init__.py

```diff
@@ -127,6 +129,8 @@
     "PytestConfigWarning",
     "PytestDeprecationWarning",
     "PytestExperimentalApiWarning",
+    "PytestRemovedIn7Warning",
```

Comment:
When we will remove the `PytestRemovedIn7Warning` class itself? In pytest 8, just as we turn `PytestRemovedIn8Warning` into an error?

Reply:
I was thinking to remove it in 7.1:

- Users of pytest should be pinning their pytest version. So if they added an ignore for `PytestRemovedIn7Warning`, once they've upgraded to 7.1 it's no longer relevant and can be removed.
- Plugins which want to support multiple pytest versions really shouldn't do `PytestRemovedIn7Warning` ignores...

Reply:
> So if they added an ignore for PytestRemovedIn7Warning, once they've upgraded to 7.1 it's no longer relevant and can be removed.

Hmm OK, just to mention that it would immediately break the test suite (the warnings module raises an error if you try to ignore an exception that doesn't exist). But of course that's just removing one line.

Even so might make sense to remove it in 8 because of that? Either way I'm fine with what you decide, just wanted to mention that bit in case it was not clear.

Reply:
If someone needed to add this line, then their testsuite will fail on 7.1 anyway because the feature they added the ignore for will have been removed. So failing fast seems better to me actually.

Reply:
> If someone needed to add this line, then their testsuite will fail on 7.1 anyway because the feature they added the ignore for will have been removed. 

Good point. 👍